### PR TITLE
NXP-28228: Retrieve Comments of document with the right permissions M…

### PR DESCRIPTION
**Pull request by RSalem07**
…ust not failed

### CODE REVIEW

**Comment by RSalem07** [comment]
T&P in Progress https://qa2.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-saouana-10.10/50/

File: nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/PropertyCommentManager.java [hunk 1]
```diff
@@ -388,8 +387,16 @@ public boolean hasFeature(Feature feature) {
...
     @Override
-    public DocumentRef getTopLevelCommentAncestor(CoreSession session, DocumentRef commentIdRef) {
-        return getAncestorRef(session, session.getDocument(commentIdRef));
+    public DocumentRef getTopLevelCommentAncestor(CoreSession s, DocumentRef commentIdRef) {
+        DocumentModel documentModel = CoreInstance.doPrivileged(s, session -> {
+            if (!session.exists(commentIdRef)) {
+                throw new CommentNotFoundException(String.format("The comment %s does not exist.", commentIdRef));
+            }
+
+            return session.getDocument(commentIdRef);
```
[the remark this answers is not in the document]
Comment (RSalem07) [reply nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/PropertyCommentManager.java]: No, the exiting one before i did the refactoring didn't detached it.

```
protected DocumentRef getAncestorRef(CoreSession session, DocumentModel documentModel) {
        return CoreInstance.doPrivileged(session, s -> {
            if (!documentModel.hasSchema(COMMENT_SCHEMA)) {
                return documentModel.getRef();
            }
            DocumentModel ancestorComment = getThreadForComment(s, documentModel);
            return new IdRef((String) ancestorComment.getPropertyValue(COMMENT_PARENT_ID));
        });
    }
```

[the remark this answers is not in the document]
Comment (troger) [reply nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/PropertyCommentManager.java]: This code is returning a `DocumentRef` not a `DocumentModel`, so yes, no need to detach a ref :)

[the remark this answers is not in the document]
Comment (RSalem07) [reply nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/PropertyCommentManager.java]: yes but i was talking about `DocumentModel ancestorComment`  they didn't detach it before my new method

**Pull request merged**